peekaboo/toolbox/files.py

```diff
@@ -37,16 +37,40 @@
 def guess_mime_type_from_file_contents(file_path):
     """  Get type from file magic bytes. """
     mt = magic.from_file(file_path, mime=True)
-    if mt:
-        return mt
+    if not mt:
+        return None
+
+    return mt
+
+
+def register_additional_mime_types():
+    # do not re-register if already done, take the module's inited flag as
+    # indicator
+    if mimetypes.inited:
+        return
+
+    # will init the module and set inited flag on first call
+    mimetypes.add_type('application/javascript', '.jse')
```

Comment:
Can we use the same mechanism to take care of the .ksh extension for text/plain?

Reply:
Unfortunately no because this only affects the forward lookup from file extension to mime type. When using guess_extension() the returned file extension is non-deterministic as stated in the comment in guess_file_extension_from_mime_type() below. Which kinda makes sense because if you only know a mime type and there's multiple kinds of input files with different extensions and applications handling them, any file extension is as good as the next.
What I guess we'd actually need to do is submit them to Cuckoo with *all* known file extensions for that type to see if they trigger anything with any of them, rationale being that the user's mail program may have different associations for that mime type with different programs e.g. on different workstations and therefore cause different actions.

Reply:
Wouldn't submission with profile 'generic' produce that exact behaviour?
`cmd start`

Reply:
This seems to be a bigger discussion. Let's defer it to #47. I will remove this change from the changeset and repurpose the PR so we get all the other cleanup without this particular, non-finalized change.